apps/src/music/views/TimelineSimple2Events.jsx

```diff
@@ -0,0 +1,142 @@
+ */
+const TimelineSimple2Events = ({
+  currentPlayheadPosition,
+  barWidth,
+  eventVerticalSpace,
+  getEventHeight
+}) => {
+  const playerUtils = useContext(PlayerUtilsContext);
+  const soundEvents = playerUtils.getSoundEvents();
+
+  const getVerticalOffsetForEventId = id => {
+    return (
+      getUniqueIndexForEventId(id) * getEventHeight(currentUniqueSounds.length)
+    );
+  };
+
+  const getUniqueIndexForEventId = id => {
+    return currentUniqueSounds.indexOf(id);
+  };
+
+  // Generate a list of unique sounds, with uniqueness being a combination of
+  // the function name and the sound ID.
+  // Let's cache the value of currentUniqueSounds so that the various helpers
+  // we call during render don't need to recalculate it.  This also ensures
+  // that we recalculate unique sounds, even when there are no entries to
+  // render.
+  const currentUniqueSounds = [];
+  for (const songEvent of soundEvents) {
+    const id = songEvent.functionContext.name + ' ' + songEvent.id;
+    if (currentUniqueSounds.indexOf(id) === -1) {
+      currentUniqueSounds.push(id);
+    }
+  }
+
+  // Next, go through all sound events, and for each unique function that
+  // is involved, adjust the boundaries of timeline space if necessary.
+  // The outcome is an array of functions that generate sounds, with the
+  // timeline boundaries for each.
+  // Each timeline boundary has left/right position in measures, and
+  // top/bottom position in pixels.
+  const uniqueFunctionExtents = [];
+  for (const soundEvent of soundEvents) {
+    const soundId = soundEvent.id;
+    const functionName = soundEvent.functionContext.name;
+    const length = playerUtils.getLengthForId(soundId);
+    const positionLeft = soundEvent.when;
+    const positionRight = positionLeft + length;
+    const positionTop = getVerticalOffsetForEventId(
+      functionName + ' ' + soundId
+    );
+    const positionBottom =
+      positionTop + getEventHeight(currentUniqueSounds.length);
+
+    const uniqueFunctionIndex = uniqueFunctionExtents.findIndex(
+      item =>
+        item.id === functionName + soundEvent.functionContext.uniqueInvocationId
+    );
+    if (uniqueFunctionIndex === -1) {
+      uniqueFunctionExtents.push({
+        id: functionName + soundEvent.functionContext.uniqueInvocationId,
+        positionLeft: positionLeft,
+        positionRight: positionRight,
+        positionTop: positionTop,
+        positionBottom: positionBottom
+      });
+    } else {
+      const item = uniqueFunctionExtents[uniqueFunctionIndex];
+      uniqueFunctionExtents[uniqueFunctionIndex] = {
+        id: item.id,
+        positionLeft: Math.min(item.positionLeft, positionLeft),
+        positionRight: Math.max(item.positionRight, positionRight),
+        positionTop: Math.min(item.positionTop, positionTop),
+        positionBottom: Math.max(item.positionBottom, positionBottom)
+      };
+    }
+  }
+
+  return (
+    <div style={{position: 'relative'}}>
```

Comment:
Do you want to move these inline styles into CSS? I know this code is somewhat temporary as we iterate on the design

Reply:
I think eventually they should, but I think while iterating it's handy to keep it all in one place.